optimizer.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, print_function
+
+import sys
+
+import optuna
+import absl.app
+from ds_ctcdecoder import Scorer
+import tensorflow.compat.v1 as tfv1
+
+from DeepSpeech import create_model
+from evaluate import evaluate
+from util.config import Config, initialize_globals
+from util.flags import create_flags, FLAGS
+from util.logging import log_error
+from util.evaluate_tools import wer_cer_batch
+
+
+def is_character_based():
+    character_based = False
+    if FLAGS.scorer_path:
+        scorer = Scorer(FLAGS.lm_alpha, FLAGS.lm_beta, FLAGS.scorer_path, Config.alphabet)
+        character_based = scorer.is_utf8_mode
+    return character_based
+
+CHARACTER_BASED = is_character_based()
```

Comment:
This needs to be moved into the main() function after the initialize so that flags are initialized

Reply:
Thanks. I'll move this around.